metricflow/dataflow/builder/dataflow_plan_builder.py

```diff
@@ -372,6 +377,16 @@ def _build_aggregated_conversion_node(
             predicate_pushdown_state=disabled_pushdown_state,
             measure_recipe=recipe_with_join_conversion_source_node,
         )
+        if not extraneous_linkable_specs.is_subset_of(queried_linkable_specs):
```

Comment:
Don't we need to this node BEFORE aggregating the measure?

Reply:
Otherwise we will aggregate with the filter column that we don't want included.
Can we write tests for the bug scenario (rendering & check query) to make sure this fixes it?

Reply:
removed this btw, and as for tests, i wrote them here and they include ones that had the bug https://github.com/dbt-labs/metricflow/pull/1381/commits/260148903cae5ace20539823092a24c74ae841d1